lib/src/services/jira.ts

```diff
@@ -146,7 +183,212 @@ export const Issue = T.type({
   ),
 });
 
-export type Issue = T.TypeOf<typeof Issue>;
+export const CloudJiraIssue = T.type({
+  key: T.string,
+  self: T.string,
+  fields: T.intersection([
+    T.type({
+      summary: T.string,
+      description: nullOrMissingToUndefined(T.string),
+      created: readOnlyDateFromISOString,
+      project: T.type({
+        key: T.string,
+      }),
+      timetracking: ITT.fromNullable(
+        T.partial({
+          originalEstimateSeconds: T.number,
+          timeSpentSeconds: T.number,
+          remainingEstimateSeconds: T.number,
+          originalEstimate: T.string,
+          timeSpent: T.string,
+        }),
+        {
+          originalEstimateSeconds: 0,
+          timeSpentSeconds: 0,
+          remainingEstimateSeconds: 0,
+          originalEstimate: "0d",
+          timeSpent: "0d",
+        }
+      ),
+      fixVersions: T.readonlyArray(
+        T.type({
+          id: T.string,
+          name: T.string,
+          released: T.boolean,
+        })
+      ),
+      aggregateprogress: T.type({
+        progress: nullOrMissingToUndefined(T.number),
+        total: nullOrMissingToUndefined(T.number),
+        percent: nullOrMissingToUndefined(T.number),
+      }),
+      issuetype: T.type({
+        name: T.string,
+        subtask: T.boolean,
+      }),
+      assignee: T.type({
+        accountId: T.string,
+        displayName: T.string,
+      }),
+      status: T.type({
+        id: T.string,
+        name: T.string,
+        statusCategory: T.type({
+          id: nullOrMissingToUndefined(T.number),
+          name: nullOrMissingToUndefined(T.string),
+          colorName: nullOrMissingToUndefined(T.string),
+        }),
+      }),
+      comment: nullOrMissingToUndefined(
+        T.readonly(
+          T.intersection([
+            PaginatedResults,
+            T.type({
+              comments: T.readonlyArray(IssueComment),
+            }),
+          ])
+        )
+      ),
+      worklog: nullOrMissingToUndefined(
+        T.readonly(
+          T.intersection([
+            PaginatedResults,
+            T.type({
+              worklogs: T.readonlyArray(IssueWorklog),
+            }),
+          ])
+        )
+      ),
+      duedate: nullOrMissingToUndefined(ITT.DateFromISOString),
+    }),
+    T.partial({
+      aggregatetimeestimate: nullOrMissingToUndefined(T.number),
+      aggregatetimeoriginalestimate: nullOrMissingToUndefined(T.number),
+      aggregatetimespent: nullOrMissingToUndefined(T.number),
+      parent: T.type({
+        id: T.string,
+        key: T.string,
+      }),
+    }),
+    // Required to model custom fields whose names cannot be known at compile time
+    T.readonly(T.record(T.string, T.unknown)),
+  ]),
+  changelog: ITT.fromNullable(
+    T.type({
+      histories: T.readonlyArray(ChangeLog),
+    }),
+    {
+      histories: [],
+    }
+  ),
+});
+
+/*Type that can represent both On-prem Jira and Cloud Jira */
+export const GenericJiraIssue = T.type({
```

Comment:
What is the actual difference in these types? The is a lot of similarity between the three types.

Reply:
The main difference is that the `GenericJiraIssue` type works with both on-prem and cloud formats. The data that gets sent over from cloud jira and on-prem jira are different enough that they have to get coded to something generic. But the generic issue doesn't need to be converted back (or that's what I assume - I still have not tested that aspect yet)

Reply:
Please extract everything that is common to a different type that is then included into the other types using `intersect` so that:
1. it is immediately obvious what the differences are
2. we have less duplication

Reply:
Completed